Sources/Bow/Data/Sum.swift

```diff
@@ -0,0 +1,105 @@
+import Foundation
+
+public class ForSum {}
+public typealias SumOf<F, G, V> = Kind3<ForSum, F, G, V>
+public typealias SumPartial<F, G> = Kind2<ForSum, F, G>
+
+public enum Side {
+    case left
+    case right
+}
+
+public class Sum<F, G, V> : SumOf<F, G, V> {
```

Comment:
Down the road, we may have to change the name of this data type here and in Arrow since it is definitely confusing when thinking about the `Sum` `Monoid<Number>` which we still need a strategy to disambiguate from the `Product` one

Reply:
I agree, it was confusing to me when I ported it from Arrow.